Dockerfile

```diff
@@ -0,0 +1,15 @@
+FROM python:3.12.3-alpine3.19
+
+WORKDIR /app
+
+COPY requirements.txt /app
+
+RUN apk update && \
+    apk upgrade && \
+    pip install --no-cache-dir -r requirements.txt
```

Comment:
Any particular reason you have coupled update, upgrade and pip install together?

Reply:
No extra build layer which help to improve efficiency of dockerfile.

Reply:
If something gets changed in requirements.txt, it will run this whole step again; updating, upgrading alpine packages and installing python packages again. Wouldn't it be better to move apk update and apk upgrade few layers above to better use docker caching feature?